### DISCUSSION TITLE
Need help configuring PrismaORM in a serverless project (AWS Lambda and AWS SAM)

### DISCUSSION THREAD

**Discussion by lucasdavidferrero** [prisma/prisma#25223] [Q&A] [answered]
### Question

I have development issues (linting, not getting types) when using PrismaORM in lambdas. My project has a layer that is used in every other lambda. Layer is located at: /src/layers/prisma:
![2024-09-13_11h51_04](https://github.com/user-attachments/assets/90b7b6aa-f52f-4aba-a6a6-4a3dc9f363ad)

The other lambdas - that are used to capture http requests- are divided by modules. For example there are two root modules: /src/modules/Processes and /src/modules/Products. Inside each module I have lambdas, a folder for services, typescript classes, and so on. I'd like to get from those module's directory the prisma's types and also the prisma client to interact with the database.   How should I use prisma that was defined in a layer in other lambdas? I already have configured the template.yaml that AWS SAM requires for infrastructure provision (Layer is referenced in all lambdas resources). 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Windows
- Database: MySQL
prisma: 5.19.1
@prisma/client: 5.19.1
Operating System: win32
Architecture: x64
Node.js: v20.17.0

**Comment by ludralph** [comment, marked as answer]
Hi @lucasdavidferrero 

You may find this [article](https://medium.com/ama-tech-blog/deploying-prisma-with-aws-lambda-layers-in-cdk-10608f5598c5) useful for configuring Prisma with AWS lambda layers. We also have a [feature request](https://github.com/prisma/prisma/issues/12506) to support this by default. You may also find this closed [issue](https://github.com/prisma/prisma/issues/8337) useful for your usecase.
